### PR TITLE
feat: MimeMessage: Support extra Content-Type directives

### PR DESCRIPTION
**Pull request by iequidoo**
Support extra "Content-Type" directives in MimeMessage passed in a fake "Content-Type-Deltachat-Directives" header which is then removed from the resulting message. This is needed to support adding "protected-headers=\"v1\"" directive to the "Content-Type" header of multipart messages.

This is related to https://github.com/deltachat/deltachat-core-rust/issues/2302

### CODE REVIEW

**Comment by iequidoo** [comment]
@link2xt, which branch should i merge this to, not to break the current versions of DC? Do we have an elaborated procedure for that?

**Comment by link2xt** [comment]
Why would it break existing version? Isn't the change backwards-compatible if nobody sets this header?

There is basically only the `master` branch.

File: src/message.rs [hunk 1]
```diff
@@ -176,38 +176,39 @@ impl MimeMessage {
...
-
+        // Since a single header can not be replace, clone all the headers
+        // except th `Content-Type`, which needs will be added later.
+        let mut headers = self.headers_without_content_type();
+        // Instead of returning the value of the `Content-Type` as a tuple,
+        // return it as String since it could have params.
+        let mut ct_value: String;
+        if let Some(message_type) = self.message_type {
+            // Use `message_type` as `Content-Type` if present
+            ct_value = message_type.to_string();
```
Comment (iequidoo) [inline src/message.rs:187]: @link2xt you're right, it's backward-compatible, we even never get here if `self.children.len() == 0`

File: src/message.rs [hunk 2]
```diff
@@ -176,38 +176,39 @@ impl MimeMessage {
...
+        } else {
+            ct_value = match self.headers.get("Content-Type".to_string()) {
+                // Use `Content-Type` header if present
+                Some(ct_header) => ct_header.get_value().unwrap(),
+                // Or assume it is `multipart/mixed`.
+                None => MimeMultipartType::Mixed.to_string()
+            };
+        }
+        if let Some(directives) = self.headers.get("Content-Type-Deltachat-Directives".to_string()) {
+            let directives: String = directives.get_value().unwrap();
```
Comment (link2xt) [inline src/message.rs:197]: This `unwrap` looks a bit scary, but `FromHeader` for `String` never returns error and there is a similar pattern in other places of the code, so it's ok. Could still wrap this into `if let Ok(..)` for extra safety, dealing with panics especially on mobile is hard, better avoid them when we can.

**Review by link2xt — APPROVED** [review]

File: src/message.rs (hunk 2 above)
Comment (iequidoo) [reply src/message.rs:197]: This function already did the same for "Content-Type" header, so i decided not to change this right now. But i agree that better to refactor this

**Pull request merged**